pkg/routeagent_driver/handlers/ovn/non_gateway_route_handler.go

```diff
@@ -85,13 +85,18 @@ func (h *NonGatewayRouteHandler) RemoteEndpointCreated(endpoint *submarinerv1.En
 		return nil
 	}
 
-	_, err := h.smClient.SubmarinerV1().
-		NonGatewayRoutes(endpoint.Namespace).Create(context.TODO(),
-		h.newNonGatewayRoute(endpoint), metav1.CreateOptions{})
-	if err != nil && !apierrors.IsAlreadyExists(err) {
+	logger.V(log.TRACE).Infof("Remote endpoint %q created event received on gateway node", endpoint.Name)
+
+	ngwr := h.newNonGatewayRoute(endpoint)
+
+	result, err := util.CreateOrUpdate(context.TODO(), NonGatewayResourceInterface(h.smClient, endpoint.Namespace),
+		ngwr, util.Replace(ngwr))
+	if err != nil {
 		return errors.Wrapf(err, "error processing the remote endpoint create event for %q", endpoint.Name)
 	}
 
+	logger.V(log.TRACE).Infof("NonGatewayRoute %s: %#v", result, ngwr)
```

Comment:
Instead of logging the entire ngwr, can we simply log the ngwr.RoutePolicySpec alone?